### PR TITLE
Fixing "No exported functions in windows dll" issue

### PR DESCRIPTION
**Pull request by msmshazan**
Patch is related to issue https://github.com/cwida/duckdb/issues/934
This patch only exposes the C API of duckdb via DLL

### CODE REVIEW

**Comment by hannesmuehleisen** [comment]
Looks great, thank you. Will merge once tests pass.

**Comment by hannesmuehleisen** [comment]
Nevermind the R OSX fix its a bit flaky. I restarted it.

File: src/include/duckdb.h [hunk 1]
```diff
@@ -14,6 +14,14 @@
 #include <stdint.h>
 #include <stdlib.h>
 
+#ifdef _WIN32
+#define DUCKDBAPI __declspec(dllexport)
+#define DUCKDBCALL __cdecl
```
Comment (hannesmuehleisen) [inline src/include/duckdb.h:19]: One question: What does this `#define` achieve? It seems unused below.

**Comment by msmshazan** [comment]
DUCKDBCALL is for exporting function pointers if they get added to the C API in the future

File: src/include/duckdb.h (hunk 1 above)
Comment (msmshazan) [reply src/include/duckdb.h:19]: DUCKDBCALL is for exporting function pointers if they get added to the C API in the future

**Pull request merged**